_gsocproposals/2018/proposal_LHCbHDF5.md

```diff
@@ -0,0 +1,60 @@
+
+The GSoC participant will integrate the HDF file format in the Gaudi framework as a possible output for these user-level data analysis tools, thus helping the integration of the python scientific computing tools in the day-to-day arsenal of HEP physicists, and opening the way for more streamlined analysis workflows. Several compression algorithms for HDF file format (blosc, gzip) can be benchmarked and compared with the original ROOT format to choose the most suitable one. If time allows, an implementation of the [Apache Parquet](https://parquet.apache.org/) data format, also compatible with pandas, could also be implemented and benchmarked against HDF.
+
+This work has far reaching implications, as the Gaudi framework is used, besides LHCb, by the [ATLAS](http://atlas.web.cern.ch/Atlas/) collaboration at CERN, and several other experiments including [HARP](http://harp.web.cern.ch/harp/) or [BESIII](http://bes3.ihep.ac.cn/).
+
+
+## Task ideas
+
+ * Understand the kind of data HEP physicists produce and what is a possible model for storage in HDF format
+ * Learn how data conversion services are implemented in Gaudi
+ * Integrate the HDF C++ API into Gaudi
+ * Create a conversion service for the output of user-level analysis data
+ * Create dedicated tests and usage example
+ * Benchmark the implementation of the HDF format and its compression options vs the standard ROOT implementation
+ * (Optional) Implement the Apache Parquet file format and benchmark it against HDF and ROOT
+ 
+## Expected results
+
+ * Working implementation, along with tests and documentation
+ * Performance comparison of HDF vs ROOT with several compression options. If implemented, add Apache Parquet to the comparison.
+ * Public release of this extension to the Gaudi framework
+
+
+## Requirements
+
+* C++ expertise
+* Working experience with git
+* Experience with HDF5 and Apache Parquet (optional) would be an advantage
+
+## Mentors
+
+* [Albert Puig](mailto:albert.puig@cern.ch)
+* [Eduardo Rodrigues](mailto:Eduardo.Rodrigues@cern.ch)
+
+
+## Links
+ * [Gaudi](http://gaudi.web.cern.ch/gaudi/)
+ * [hdfgroup.org](https://www.hdfgroup.org/)
+
+
+## Corresponding project
```

Comment:
@hegner I believe this section is added automatically? So no need to be in the source .md file